### DISCUSSION TITLE
How to avoid afterimage when expanding/collapsing items at the same time

### DISCUSSION THREAD

**Discussion by XSAM** [TanStack/virtual#454] [Q&A] [answered]
Hello,

I am getting an afterimage (A short time wrong transform value) when resizing (expanding and collapsing) multiple items at the same time.

<img width="378" alt="image" src="https://user-images.githubusercontent.com/11306772/209530108-e38d659c-1817-424e-9b85-67c868a3cfe9.png">

Here is the screen recording: 

https://user-images.githubusercontent.com/11306772/209530316-cab0d8db-d7f2-497a-97ae-068fc0605a52.mp4

Here is the code:

https://codesandbox.io/s/dank-haze-ltc0gt?file=/src/main.tsx

---

Any ideas to remove such an afterimage?

**Comment by piecyk** [comment, marked as answer]
Yes, it could happen, as update the position of virtual item happens after the size change. One solution could be to move the whole items as one block, something like https://codesandbox.io/s/xenodochial-flower-kg35e0?file=/src/main.tsx

**Reply by XSAM** [reply]
Thanks for the reply. It works!

Should this approach become the default one to create a dynamic list? It seems less flawed.
